spec/db/db_spec.rb

```diff
@@ -51,7 +51,7 @@
   it "adds one contact to the database" do
     db.add(contact)
     expect(db.size).to eq(1)
-    expect(writer).to have_received(:run).once
+    expect(writer).to have_received(:write).once
```

Comment:
With this test, I'm not sure that what write received is being tested, which seems important